### ISSUE TITLE
Collapsable Right Nav + "Edit on GitHub" Link

### ISSUE THREAD

**Issue by thedavidprice** [redwoodjs/redwoodjs.com#178]
[GitBook](https://www.gitbook.com/) has a nice responsive UI for the right sidebar Nav. The "edit on github" link is a helpful feature as well.

See an example here: https://basarat.gitbook.io/typescript/getting-started

### Full width
<img width="927" alt="Screen Shot 2020-06-07 at 3 50 34 PM" src="https://user-images.githubusercontent.com/2951/83981918-275c0b00-a8d7-11ea-8ccf-7b376b8fe099.png">

### Collapsed
<img width="747" alt="Screen Shot 2020-06-07 at 3 50 23 PM" src="https://user-images.githubusercontent.com/2951/83981930-3c389e80-a8d7-11ea-8106-ab220ba73b08.png">

**Comment by sjmiddlebrook** [comment]
@cannikin @thedavidprice Hi, I'm new to Redwood and looking for a first issue to take on and get involved. I'd be happy to work on this one.

**Comment by thedavidprice** [comment]
@sjmiddlebrook huge thanks in advance for offering to help! 🚀

I definitely +1 this. @cannikin as our, ahem, WebMaster9000™ what do you think/suggest?

**Comment by cannikin** [comment]
So right now the responsive layout goes:

Small-Medium: Hamburger menu containing main nav
Large: Main nav at left
X-Large: Main nav at left + "On this page" nav on right

@thedavidprice When you suggested this were you thinking this is so that we can include the "On this page" nav on large screens, or small and medium as well? Shouldn't be too bad on large screens, but on small/medium we're talking double hamburgers, or nested navs or something...could get weird.

**Comment by thedavidprice** [comment]
Correct, I'm proposing to use this for small <--> large screens. Per example:
https://basarat.gitbook.io/typescript/getting-started

Agreed we don't want "stacked" hamburgers, but if we want to keep main nav hamburger on right, we could swap icons or even use text, e.g. "^Sections", "^On this Page". I think our current main nav hamburger (currently _only_ containing the left column nav) would actually work quite well on the left.

(**Note:** our current hamburger doesn't contain the top nav links, e.g. forum, discourse, etc. --> seems like we should include those as well... if in agreement let's open a separate issue.)

**Here's the example on iPhone X size emulator**
To me, this UI is very comprehensible.

<img width="381" alt="Screen Shot 2020-06-23 at 1 44 34 PM" src="https://user-images.githubusercontent.com/2951/85460662-553c8300-b558-11ea-84d8-40c91bec111d.png">

<img width="379" alt="Screen Shot 2020-06-23 at 1 44 52 PM" src="https://user-images.githubusercontent.com/2951/85460578-3807b480-b558-11ea-880f-a10b46df66e5.png">
